src/mmw/apps/modeling/mapshed/tasks.py

```diff
@@ -185,17 +196,119 @@ def nlcd_streams(sjs_result):
     }
 
 
+def get_lu_index(nlcd):
+    if nlcd == 81:
+        lu_index = 1
+    elif nlcd == 82:
+        lu_index = 2
+    elif nlcd in [41, 42, 43]:
+        lu_index = 3
+    elif nlcd in [90, 95]:
+        lu_index = 4
+    elif nlcd in [21, 71]:
+        lu_index = 7
+    elif nlcd in [12, 31]:
+        lu_index = 8
+    elif nlcd == 22:
+        lu_index = 10
+    elif nlcd == 23:
+        lu_index = 11
+    elif nlcd == 24:
+        lu_index = 12
+    elif nlcd == 11:
+        lu_index = 16
+    else:
+        return None
+
+    return lu_index - 1
+
+
+@shared_task(throws=Exception)
+def nlcd_slope_kfactor(result):
+    if 'error' in result:
+        raise Exception('[nlcd_slope] {}'.format(result['error']))
+
+    result = parse_sjs_result(result)
+
+    ag_slope_3_count = 0
+    ag_slope_3_8_count = 0
+    ag_count = 0
+    total_count = 0
+
+    slope_sum = 0
+    slope_count = 0
+
+    kf_sums = [0.0] * NLU
+    kf_counts = [0.0] * NLU
+
+    for (nlcd_code, slope, kfactor), count in result.iteritems():
+        if nlcd_code in AG_NLCD_CODES:
+            if slope > 3:
+                ag_slope_3_count += count
+            if slope > 3 and slope < 8:
+                ag_slope_3_8_count += count
+            ag_count += count
+
+        if slope != NO_DATA_VAL and slope < 500 and slope > -500:
+            slope_sum += (slope * count)
+            slope_count += count
+
+        lu_ind = get_lu_index(nlcd_code)
+        if lu_ind and kfactor != NO_DATA_VAL:
+            kf_sums[lu_ind] += (kfactor * count)
+            kf_counts[lu_ind] += count
+
+        total_count += count
+
+    # percent of AOI that is agricultural with slope > 3%
+    # see Class1.vb#7223
+    ag_slope_3_pct = float(ag_slope_3_count) / total_count
+    # percent of AOI that is agricultural with 3% < slope < 8%
+    ag_slope_3_8_pct = float(ag_slope_3_8_count) / total_count
+
+    # percent of agricultural parts of AOI with slope > 3%
+    # see Class1.vb#9864
+    n41 = float(ag_slope_3_count) / ag_count
+
+    # average slope over the AOI
+    # see Class1.vb#6252
+    avg_slope = float(slope_sum) / slope_count
```

Comment:
Would recommend using the new `RasterAverage` operation for this, as this casts Floats to Ints before taking the average, likely changing the value.